lib/generateClass.js

```diff
@@ -1740,7 +1747,8 @@ function generateFromFHIRAssignment(field, fhirElement, fhirElementPath, shrElem
       if (shrType.isPrimitive && !isExtension) {
         rhs = fhirPathString;
       } else {
-        rhs = `FHIRHelper.createInstanceFromFHIR('${shrType.fqn}', ${fhirPathString}, shrId, allEntries, mappedResources, referencesOut, ${isExtension})`;
+        const fhirType = fhirElement.type[0].code;
```

Comment:
Is it safe to assume that the type at index `0` is the right type?

Reply:
Good question. Off the top of my head I'm not sure of all the reasons that a FHIR element might have multiple types, but if this was originally a choice type like `value[x]` then it's been pre-processed so that it appears to be a single choice with a single type, and that's the use case where this is most relevant. 
Not conclusive proof, but running through the current spec there are no instances where there is a second type available to choose here.

Reply:
OK. Yes, if we've already handled `[x]` types before this then I think that should be fine.